Adding DP-1.16: Ingress traffic classification and rewrite

keysightgems opened the pull request:
Readme: https://github.com/openconfig/featureprofiles/blob/main/feature/qos/otg_tests/ingress_traffic_classification_and_rewrite_test/README.md

I have raised an issue with deviations, I'm not sure if the metadata pb file should be added or will be generated automatically. Please let me know if the generated pb file should be added as part of the PR.

keysightgems commented:
> Please resolve the conflicts in this change.

Fixed the conflicts.Thanks!

keysightgems commented:
> Please check the build failures here.

Fixed build issues. Thanks!

keysightgems commented:
Hi robshakir/dplore,
Please approve the PR. Thanks!

dplore commented on feature/qos/otg_tests/ingress_traffic_classification_and_rewrite_test/metadata.textproto:
```diff
@@ -0,0 +1,19 @@
+  deviations: {
+    interface_enabled: true
+    default_network_instance: "default"
+    qos_remark_oc_unsupported : true
```
```suggestion
    qos_remark_oc_unsupported: true
```

dplore commented on proto/metadata.proto:
URL linking to the bugs for reach newly added deviation need to be added

dplore replied:
Please add the bug links for the newly added deviations.
```
    // Devices that do not support system-description config path
    // Nokia b/431929861
    bool missing_system_description_config_path = 304;
```

dplore reviewed (changes requested):
Please fix the static checks as well as address comments. 
Please also provide a link to test results log.  This can be via a new private bug link you share here or agithub gist.

ANISH-GOTTAPU commented:
The readme is updated and more tests were added, we will open the PR once we add other tests as well.

keysightgems commented:
> please fix the static analysis failures and add the bug links to the deviations. thanks

Fixed static checks and added bug details in the *.proto file. Thanks!

dplore commented:
Hi, sorry but the metadata build is now broken, please fix.

(See go/test and rundata check outputs)

keysightgems commented:
> Also, @keysightgems I sent an github invite to you which if you accept, will allow the CI checks to run without me needing to approve so you can more easily see the results after you push a commit. Please accept the invite (sent to your github account registered email address)

Accepted, Thanks!
